sophia/src/quad/stream.rs

```diff
@@ -63,22 +60,19 @@ impl<'a, I, T, E> QuadSource for I
 where
     I: Iterator<Item = Result<T, E>>,
     T: Quad<'a>,
-    E: CoercibleWith<Error> + CoercibleWith<Never> + Into<Error>,
+    E: 'static + Error + Into<SophiaError>,
 {
     type Error = E;
 
     fn in_sink<TS: QuadSink>(
         &mut self,
         sink: &mut TS,
-    ) -> CoercedResult<TS::Outcome, Self::Error, TS::Error>
-    where
-        Self::Error: CoercibleWith<TS::Error>,
-    {
+    ) -> Result<TS::Outcome, StreamError<Self::Error, TS::Error>> {
         for tr in self {
-            let t = tr?;
-            sink.feed(&t)?;
+            let t = tr.map_source_err()?;
```

Comment:
As suggested in `StreamError`, this is uselessly unidiomatic.
I would rather write
```suggestion
            let t = tr.map_err(SourceError)?;
```
and do the same with `SinkError` below